src/FSharpPlus/Extensions/Validators.fs

```diff
@@ -0,0 +1,75 @@
+﻿namespace FSharpPlus
+
+[<RequireQualifiedAccess>]
+module RequiredValidation =
+    open System
+    open FSharpPlus.Data
+    
+    let inline validate error f v =
+        if f v then
+            Success v
+        else
+            Failure [ error ]
+
+    let inline string error value =
+        validate (error value) (String.IsNullOrWhiteSpace >> not) value
+
+    let inline greaterThan error min value =
+        validate (error value) (flip (>) min) value
+        
+    let inline greaterOrEqualThan error min value =
+        validate (error value) (flip (>=) min) value
+
+    let inline email error value =
+        let check (v: string) =
+            try
+                let _ = Net.Mail.MailAddress(v)
+                true
+            with
+            | ex -> false
+
+        validate (error value) check value
+
+    let inline guid error value =
```

Comment:
this type of validator goes into the question if there is some more general class of nonEmpty restrictions?